modules/sqlite/integration/load.php

```diff
@@ -0,0 +1,150 @@
+<?php
+/**
+ * Module Name: SQLite Integration
+ * Description: Use an SQLite database instead of MySQL. <div style="background:#fff;border:1px solid #c3c4c7;border-left-width: 4px;border-left-color:#dba617;box-shadow: 0 1px 1px rgba(0, 0, 0, 0.04);padding:1em;max-width:50em;">CAUTION: Enabling this module will bring up the WordPress installation screen. You will need to reconfigure your site, and you will lose all your data. If you then disable the module, you will get back to your previous MySQL database, with all your previous data intact.</div>
+ * Experimental: Yes
+ *
+ * @package performance-lab
+ * @since x.x.x
+ */
+
+// Temporary - This will be in wp-config.php once SQLite is merged in Core.
+if ( ! defined( 'DATABASE_TYPE' ) ) {
+	define( 'DATABASE_TYPE', 'sqlite' );
+}
+
+/**
+ * Add the db.php file in wp-content.
+ *
+ * When the plugin gets merged in wp-core, this is not to be ported.
+ */
+function sqlite_plugin_copy_db_file() {
+	// Bail early if the SQLite3 class does not exist.
+	if ( ! class_exists( 'SQLite3' ) ) {
+		return;
+	}
+
+	$destination = WP_CONTENT_DIR . '/db.php';
+
+	// Bail early if the file already exists.
+	if ( file_exists( $destination ) ) {
+		return;
+	}
+
+	// Init the filesystem to allow copying the file.
+	global $wp_filesystem;
+	if ( ! $wp_filesystem ) {
+		require_once ABSPATH . '/wp-admin/includes/file.php';
+		WP_Filesystem();
+	}
+
+	if ( $wp_filesystem->touch( $destination ) ) {
+
+		// Get the db.copy file contents, replace placeholders and write it to the destination.
+		$file_contents = str_replace(
+			array(
+				'{SQLITE_IMPLEMENTATION_FOLDER_PATH}',
+				'{PERFLAB_PLUGIN}',
+				'{SQLITE_MODULE}',
+				'{PERFLAB_MODULES_SETTING}',
+			),
+			array(
+				__DIR__,
+				str_replace( WP_PLUGIN_DIR . '/', '', PERFLAB_MAIN_FILE ),
+				'sqlite/integration',
+				PERFLAB_MODULES_SETTING,
+			),
+			file_get_contents( __DIR__ . '/db.copy' )
+		);
+		$wp_filesystem->put_contents( $destination, $file_contents );
+	}
+}
+add_action( 'plugins_loaded', 'sqlite_plugin_copy_db_file' );
```

Comment:
I did it this way mostly for redundancy... If for some reason copying the file fails (for example wrong file/folder permissions in the `wp-content` folder), then the user will get an admin notice and it will try again to copy the file in the next page-load.
It was also very useful for dev purposes, 'cause we can just delete the `wp-content/db.php` file and it will copy the `db.copy` file again without the need for any additional actions - which is extremely useful as I'm constantly tweaking the `db.copy` file 😅
Performance-wise it doesn't really have an impact, because if the file exists, it bails early.

What do you think? Should we still change it to run on `pre_update_option_`? 🤔

Reply:
@aristath I think it's still cleaner to do this in a dedicated activation routine rather than checking on every page load.

What I would rather suggest as the fallback / redundancy mechanism would be to check this e.g. on `admin_notices` and display a warning if the file is missing despite the module being active. We could even add a button there like "Recreate DB file" in the future, but maybe not necessary right away.

Regarding a cleaner activation and deactivation routine, I suggested something in https://github.com/WordPress/performance/pull/547#discussion_r1004799222.

Reply:
Refactored this and now it only happens on module activation.